### ISSUE TITLE
Some unicode glyphs don’t scale linearily on the vertical axis.

### ISSUE THREAD

**Issue by phaazon** [kovidgoyal/kitty#3144]
**Describe the bug**
Some unicode characters render with the wrong “height” or get cut off when the increasing or decreasing the size font.

**To Reproduce**
1. Just echo ▁ this unicode character. The “lenny” multi-unicode glyph renders very badly too and it looks it’s linked — this one ( ͡° ͜ʖ ͡°)
2. Decrease the font size to minimal value.
3. Increase gradually.
4. Notice how the height of the glyph changes in a non-linear, stochastic way. It is even worse when you also use the vertical version — ▎— which seems okay.

**Expected behavior**
The unicode glyphs should scale linearily on the vertical axis.

**Screenshots**

Left is `kitty`, right (over it) is `alacritty`.
![Screenshot_20201204_225714](https://user-images.githubusercontent.com/506592/101219683-abbff880-3684-11eb-8250-974c4c222917.png)

Other example of how scaling is not linear when increasing the font — this one is already very visible when the font size is small, the glyph looks way too big.
![Screenshot_20201204_225112](https://user-images.githubusercontent.com/506592/101219809-dc079700-3684-11eb-8f2d-2d25bf2209c1.png)
![Screenshot_20201204_225103](https://user-images.githubusercontent.com/506592/101219810-dca02d80-3684-11eb-8986-974a245f50c2.png)

**Environment details**
OS: Linux orchid 5.9.11-arch2-1 #1 SMP PREEMPT Sat, 28 Nov 2020 02:07:22 +0000 x86_64 GNU/Linux

```
kitty 0.19.2 created by Kovid Goyal
Linux orchid 5.9.11-arch2-1 #1 SMP PREEMPT Sat, 28 Nov 2020 02:07:22 +0000 x86_64
-`
                  .o+`
                 `ooo/
                `+oooo:
               `+oooooo:
               -+oooooo+:
             `/:-:++oooo+:
            `/++++/+++++++:
           `/++++++++++++++:
          `/+++ooooooooooooo/`
         ./ooosssso++osssssso+`
        .oossssso-````/ossssss+`
       -osssssso.      :ssssssso.
      :osssssss/        osssso+++.
     /ossssssss/        +ssssooo/-
   `/ossssso+/:-        -:/+osssso+-
  `+sso+:-`                 `.-/+oso:
 `++:.                           `-/+/
 .`                                 `

 Welcome home!
LSB_VERSION=1.4
DISTRIB_ID=Arch
DISTRIB_RELEASE=rolling
DISTRIB_DESCRIPTION="Arch Linux"
Loaded config files: /home/phaazon/.config/kitty/kitty.conf
Running under: X11

Config options different from defaults:
background                 Color(red=43, green=45, blue=58)
color0                     Color(red=54, green=58, blue=78)
color1                     Color(red=236, green=114, blue=121)
color10                    Color(red=157, green=228, blue=101)
color11                    Color(red=239, green=190, blue=99)
color12                    Color(red=91, green=185, blue=252)
color13                    Color(red=218, green=124, blue=248)
color14                    Color(red=48, green=226, blue=238)
color15                    Color(red=178, green=201, blue=236)
color2                     Color(red=160, green=201, blue=128)
color3                     Color(red=222, green=185, blue=116)
color4                     Color(red=108, green=182, blue=235)
color5                     Color(red=211, green=138, blue=234)
color6                     Color(red=93, green=187, blue=193)
color7                     Color(red=197, green=205, blue=217)
color8                     Color(red=102, green=102, blue=102)
color9                     Color(red=252, green=98, blue=107)
copy_on_select             clipboard
cursor                     Color(red=91, green=185, blue=252)
cursor_stop_blinking_after 0
cursor_text_color          Color(red=255, green=255, blue=255)
enable_audio_bell          False
font_family                JetBrainsMono Nerd Font
font_size                  14.0
foreground                 Color(red=171, green=178, blue=191)
initial_window_height      (620, 'px')
initial_window_width       (840, 'px')
rectangle_select_modifiers 2
remember_window_size       False
selection_background       Color(red=64, green=64, blue=64)
selection_foreground       Color(red=234, green=234, blue=234)
strip_trailing_spaces      smart
sync_to_monitor            False
update_check_interval      24.0
window_border_width        (1.0, 'pt')
Added shortcuts:
	 shift+control+plus KeyAction(func='change_font_size', args=(True, '+', 2.0))
Changed shortcuts:
	 shift+control+equal KeyAction(func='change_font_size', args=(True, None, 0.0))
	 shift+control+s KeyAction(func='scroll_line_up', args=())

```

**Additional context**
Try to reproduce the problem with `kitty --config NONE` if you cannot then post a minimal kitty.conf that reproduces the problem. If the problem involves interaction with some other terminal program post a minimal config for that program to reproduce the problem as well.

**Comment by phaazon** [comment]
Also, I’m using the same font on `alacritty`, if that’s useful.

**Comment by kovidgoyal** [comment]
dup of #3097

**Comment by phaazon** [comment]
@kovidgoyal the issue you linked is about width (which I don’t reproduce, btw). I had a look at the ticket you closed (#3097), and… does it mean you don’t care? You don’t plan to fix this?

**Comment by Luflosi** [comment]
The other issue is closed because there is a fix. Either wait for the next version to be released or [build from source](https://sw.kovidgoyal.net/kitty/build.html). I can't comment on whether the two issues are related or not.
